source/Octopus.Tentacle/Kubernetes/KubernetesPodMonitor.cs

```diff
@@ -156,11 +179,36 @@ internal async Task OnNewEvent(WatchEventType type, V1Pod pod, CancellationToken
             }
         }
 
-        IList<ITrackedScriptPod> IKubernetesPodStatusProvider.GetAllTrackedScriptPods() =>
-            podStatusLookup.Values.Cast<ITrackedScriptPod>().ToList();
+        IList<ITrackedScriptPod> IKubernetesPodStatusProvider.GetAllTrackedScriptPods()
+        {
+            WaitForInitialLoadToFinish();
+
+            return podStatusLookup
+                .Concat(pendingPodStatusLookup)
+                .ToLookup(p => p.Key)
+                .Select(g => (ITrackedScriptPod)g.First().Value) //Deduplicate
+                .ToList();
```

Comment:
Rather than having dupes, should we remove the pod out of pending once it's been added via the API/Watch? Then we should never have duplicates?

Reply:
There's also the batch load that happens every 10min - which we would can do - just most likely needs a `lock` statement to ensure we sync the 2 dictionaries